src/coreclr/tools/aot/ILCompiler.ReadyToRun/Compiler/DependencyAnalysis/ReadyToRun/Target_Riscv64/ImportThunk.cs

```diff
@@ -0,0 +1,69 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+
+using ILCompiler.DependencyAnalysis.Riscv64;
+
+namespace ILCompiler.DependencyAnalysis.ReadyToRun
+{
+    /// <summary>
+    /// This node emits a thunk calling DelayLoad_Helper with a given instance signature
+    /// to populate its indirection cell.
+    /// </summary>
+    public partial class ImportThunk
+    {
+        protected override void EmitCode(NodeFactory factory, ref Riscv64Emitter instructionEncoder, bool relocsOnly)
+        {
+
+            switch (_thunkKind)
+            {
+                case Kind.Eager:
+                    break;
+
+                case Kind.DelayLoadHelper:
+                case Kind.VirtualStubDispatch:
+                    // T8 contains indirection cell
+
+                    if (!relocsOnly)
+                    {
+                        // ori a4, r0, #index
+                        int index = _containingImportSection.IndexFromBeginningOfArray;
+                        instructionEncoder.EmitMOV(Register.X14, checked((ushort)index));
+                    }
+
+                    // get pc
+                    // auipc T1=X6, 0
+                    instructionEncoder.EmitPC(Register.X6);
+
+                    // load Module* -> T1
+                    instructionEncoder.EmitLD(Register.X6, Register.X6, 0x18);
```

Comment:
I think it is wrong. Please update to correct offset.

Reply:
Done